### ISSUE TITLE
Incorrect new array indexing formatting

### ISSUE THREAD

**Issue by pbiggar** [fsprojects/fantomas#1985]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMBnRA5gLaIB2OAvGAMoCeuWiRAdAKoBOAlgBTRYAezAEqIAjgFdEDZgAUAhlgAWzAGpz0kgJTNqhEuVwBtAIwBdSABpwUfFiydSBXKjCHrEUBC9QAJFloADshokA5Ylu5ekD4AbuqSLm7e3gAMVslRAJKkSOTUnABeyOkZYAAskRCmkQC%2BJR6Vvv5BLqHkERnRcRrBro0QxvXJkACycvwAMg6IE2QESh2lxgBMKY3V3nWRnsN%2Bgb2QAEawsOiL3l3xvUmly0MXukSccOiwpACCWACiOQDyAGZTUjFfpgf7qfDrWpDHYXPYtELHU7nKKxK6JUEAZnuUWoATk0EQACFEP9YOxEPJ2HISIx2CjvFh2JIoZsYY1os0DkizjjLj0MaVyjiIJA8QTiaTyTNYAB3RDsaByfDZGKwJX2N4MrxMlkZDZeLbeWGormtHna-kJNA3DIAVhFUHFhJJZIprACQUVysQqvVCk4Wsd4PQkP10O2HPh3JOvKj3WtfSFADZHWL8S6pRSAMLoZW4bNvBjMvjk7UQENh5IGiBGrwm0XR82xy1ogU20EAdjTzslbsQIyYhwV5bBEMQrMN7M6TcRLb5bcTtuSAA4exm%2B9LC69SKPKxPw2zIzOzXPkQuE9dQQBOdcS97-OmFohEOSj3UH6sR41R09QC0XuiHZCsYaSgumEqutKjzPKcQagvuk61tOux-kc87xkBSZLIM4G9g%2BdIwS88FCh%2BSFgHWDQnvszbnph7bYRkKxptkuRYL8pAACrsLQADqnALMG47kZRYANk0NFnnGM6XoKSzYnhG7vOwsDiDkAAiiDoJwTx0u%2BzKft4NYUShcJoWEraycBSxlGmYz8Jk-ycYoZBfKGiDUIo5JYPx8CCaCZRrIeU7HqhklQBZgEMcu3jGA64H2dk-ycPwvxego5JfPwAQUrguCBrujp2kFX5Hj%2B1EIhF7RRUuoLGKmCXjCIcDsPAvn%2BUKgUiaZprhW04Q1VeIHdo1-DNeS8AAHLiEQw7sACmSMEQziOsY3WhWZfVEOI6D2NpwIAGLkq%2BdgKpxNGDXJTFruB42tSMO17dMR3sCdelptAihyNSfAKgA%2BrKnB%2BYokDreVYWVf1llYTFXjGLeo3KdStC-OwUwMO1IOOl1wXIRtvWQ5F9G1UKqx2eMSNyCjaOcAw02zQqC1LStdVg-Wv5bY9nD7YgL1vWdF3E0Nty4UKkCU9T6NYA9u3c89x0KO94Gfd9BJ0gDQMLGzVEQwcRMyTDoLLHco1qD0RIOPADgEJjo4riVRnfuzFV69VQtXckywKWL9kHWpfAFRbOTW7b2MO14xmieJnJ9fruxWYxnu2aN6mwFgADi2BZTlUj5W8oegsV2tiRzkMAe71kZMs8U%2B1zPNEq80AANZEj9TfYLgHHUDSiDbjNhUIcJuMmfjjZoeXBvRUbDVi5NiCyvX2DymQ51BJpyWkAJBXvDkg4M%2BwLNCohw9R6XMZ0ZPJO3CNYsANKIIgASOc5ZDZN3JBAiCR9D6VIXg5thM3aX2FlXG6PtxhuSeLgRQBdOrhyqE7HWADz7SXjobUmCMxb5EcJgZSBAZpkB8ogQ4IwEBf1KMfX%2BeN-4ExQdDKeQpMRgTFu8bSBBSB%2B1IAHN4%2BR2EKHEBSTisBWKEIDCRChP9HZlWdrrWiqC4QJ1hhATEotSji12gqUgAYYgykcHvOa68HBbyLHuSREdEElxdnI%2BhV8MiYhNrXWWn8Jg0kOPAOQuZYD5UcHPBe0xTHuWLtHWc-4MLAI9t4TE3s1HqVpnIQ4mBIG00UHQcg4wAlVikX-GRyDWhkHgACT%2B1B-CYBsSA5ImJk5ix%2BAUwE-iPrsHQP8UGJ8epjz6hPNBDDSiYhrmo%2B%2Bj8RHkGyK3OQXCsaD0Ca00eEky5hK6bYipM81ENzGU3T%2BuBlKqRyHPBgiB4AyyesCbOuU86kEPqUMi0yaHtLmRfBZ5TIk3xWd9KCFJYm4GgFwJ4WjGDwFYJvN4mloB5mpJqAe38plUJHjc2ZdDLqVwqWAtRxSuB8FIUgDJhlzGbGsKYEooRcAHXyi4RCNQgA)

#### Code

```fsharp
let segment = System.Uri(ctx.Request.Path.Value).Segments[1]
```
            

#### Result

```fsharp
let segment = System.Uri(ctx.Request.Path.Value).Segments [ 1 ]

```
            
#### Problem description

After formatting, there is a compiler error: "error FS3217: This expression is not a function and cannot be applied. Did you intend to access the indexer via 'expr[index]'"

#### Extra information

- [x] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas 4.6 branch at 11/24/2021 18:43:36 - 9eb1365627a87929af770bd60baed8be76e61103

Default Fantomas configuration

**Comment by nojaf** [comment]
Thanks for this, good spot.
`IndexWithoutDotExpr` should be higher in the genExpr pattern match I believe.
Will check maybe later this week for this.

**Comment by nojaf** [comment]
Fix available in 4.6

**Comment by pbiggar** [comment]
Great, thank you!
